cirq-core/cirq/value/condition.py

```diff
@@ -146,23 +162,52 @@ def parse_sympy_condition(s: str) -> 'cirq.SympyCondition':
     The measurement keys in a sympy condition string must be wrapped in curly
     braces to denote them. For example, to create an expression that checks if
     measurement A was greater than measurement B, the proper syntax is
-    `cirq.parse_sympy_condition('{A} > {B}')`."""
-    keys_to_indexes: Dict[str, str] = {}
-
-    def replace(m):
-        g1 = m.group(1)
-        if g1 not in keys_to_indexes:
-            keys_to_indexes[g1] = f'x{len(keys_to_indexes)}'
-        return keys_to_indexes[g1]
-
-    result = re.sub(r'{([^}]+)}', replace, s)
-    expr = sympy.sympify(result)
-    keys = [measurement_key.MeasurementKey.parse_serialized(key) for key in keys_to_indexes.keys()]
+    `cirq.parse_sympy_condition('{A} > {B}')`.
+
+    A backslash can be used to treat the subsequent character as a literal
```

Comment:
The more I think about this, the more I get the sense that pointing to the `SympyCondition` constructor would be preferable. The way I see it, this method is a convenient shorthand for simple conditions; for more complex circuits, users will likely need to refer to the `MeasurementKey` objects in their conditions, at which point the constructor is preferable.

Reply:
Actually I didn't know until Matt Neely pointed out on the other thread, that sympy symbols can actually have arbitrary names. So that makes me think that maybe this parsing is unnecessary. Users can create and use the sympy expressions directly. That way we don't have to maintain the mapping either (unless `with_keys` has problems with special characters). The one caveat is that the result of `str(condition)` might not be parseable if there are special names in the symbols, though that should be expected.

So, instead of all this parsing, we just have SympyCondition take a sympy expression in the constructer, and leave it up to the users to create that expression. What do you think about that approach?

Reply:
I went ahead and made that change. I think this is pretty good. Is someone clamors for a parsing function then we can do that later.